docs/developers/README.md

```diff
@@ -155,3 +156,24 @@ such as the following:
 - The environment variable HOME is set to `/builder/home`, used by the builder
   tools and injected on into all of the step containers
 - Default location for output-images `/builder/output-images`
+
+## Handling of injected sidecars
+
+The [Downward API](TODO: link) is used to project an annotation on the
+TaskRun's Pod into the entrypoint container as a file.
+
+The annotation starts as an empty string, so the file projected by the downward
+API has zero length. The entrypointer spins, waiting for that file to have
+non-zero size.
+
+The sidecar containers start up. Once they're all in a ready state, the
+annotation is populated with string "READY", which in turn populates the
+Downward API projected file. The entrypoint binary recognizes
+that the projected file has a non-zero size and allows the Task's steps to
+begin.
+
+On completion of all steps in a Task the TaskRun reconciler stops any
+sidecar containers. The `Image` field of any sidecar containers is swapped
+to the nop image. Kubernetes observes the change and relaunches the container
+with updated container image. The nop container image exits. The container
+is considered
```

Comment:
awesome!! thanks for writing this up, very clear! and a pretty nice design imo :D (*high five* @dicarlo2 :D)